### PR TITLE
DEP: add deprecation warning for vertices attribute in qhull

### LINKED ISSUE #16255: DEP: add deprecation warning for `vertices` kwarg in qhull
Deprecated in documentation only since https://github.com/scipy/scipy/pull/377, specifically https://github.com/scipy/scipy/commit/df5813b27bd6049a5b426dd1a1311f1d29892528.

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

closes #16255
#### What does this implement/fix?

The attribute `vertices` was deprecated in documentation only. This pr adds a deprecation warning and test with details for removal.

### CODE REVIEW

**Review by h-vetinari — APPROVED** [review]
You're amazing - thanks so much for tackling this so quickly! :)

File: scipy/spatial/_qhull.pyx [hunk 1]
```diff
@@ -1859,10 +1864,17 @@ class Delaunay(_QhullUser):
...
         # Backwards compatibility (Scipy < 0.12.0)
-        self.vertices = self.simplices
+        self._vertices = self.simplices
 
         _QhullUser._update(self, qhull)
 
+    @property
+    def vertices(self):
+        msg = ("Delaunay attribute 'vertices' is deprecated in favour of "
+        "'simplices' and will be removed in Scipy 1.11.0.")
```
Comment (tupui) [inline scipy/spatial/_qhull.pyx:1874]: I am surprised that this was not flagged by the linter.
```suggestion
        msg = ("Delaunay attribute 'vertices' is deprecated in favour of "
               "'simplices' and will be removed in Scipy 1.11.0.")
```

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you @j-bowhay. CI failure seems related.

@j-bowhay @h-vetinari Thank you both for all the work on deprecations. If you are available tomorrow we have our community meeting. Would be nice to meet you and discuss your efforts 😃

**Comment by h-vetinari** [comment]
> @j-bowhay @h-vetinari Thank you both for all the work on deprecations. If you are available tomorrow we have our community meeting. Would be nice to meet you and discuss your efforts 😃

Thanks for the invite! I'll be in a train at that [time](https://mail.python.org/archives/list/scipy-dev@python.org/message/URJX2TJO2KVRSNVR5YSGMXSLUHE2ALJ7/), but I'll try to join if reception permits.

**Comment by tupui** [comment]
No worries. You are always welcome to join to these meetings. They are for the whole community.

File: scipy/spatial/_qhull.pyx [hunk 2]
```diff
@@ -1720,6 +1721,10 @@ class Delaunay(_QhullUser):
         .. versionadded:: 0.12.0
     vertices
         Same as `simplices`, but deprecated.
+
+        .. deprecated:: 0.12.0
+        Delaunay attribute `vertices` is deprecated in favour of `simplices`
```
[the remark this answers is not in the document]
Comment (j-bowhay) [reply scipy/spatial/_qhull.pyx:1726]: ahh always screw this up

**Comment by j-bowhay** [comment]
> @j-bowhay @h-vetinari Thank you both for all the work on deprecations. If you are available tomorrow we have our community meeting. Would be nice to meet you and discuss your efforts smiley

Unfortunately, I have a presentation training course (ugh) tomorrow so wont be able to make it

**Review by tupui — APPROVED** [review]
LGTM now, thanks again.

**Pull request merged**